src/core/model/table/typed_column_data.h

```diff
@@ -173,16 +174,20 @@ class TypedColumnDataFactory {
 
     inline static std::vector<TypeId> const kAllCandidateTypes = {
             +TypeId::kDate, +TypeId::kInt, +TypeId::kBigInt, +TypeId::kDouble, +TypeId::kString};
-    inline static std::unordered_map<TypeId, std::regex> const kTypeIdToRegex = {
-            {+TypeId::kBigInt, std::regex(R"(^(\+|-)?\d{20,}$)")},
-            {+TypeId::kInt, std::regex(R"(^(\+|-)?\d{1,19}$)")},
-            {+TypeId::kNull, std::regex(Null::kValue.data())},
-            {+TypeId::kEmpty, std::regex(R"(^$)")}};
+    inline static std::unordered_map<TypeId, boost::regex> const kTypeIdToRegex = {
+            {TypeId::kDate,
+             boost::regex(
+                     R"(^([0-9]{4})[-.\/]?(1[0-2]|0[1-9]|[1-9])[-.\/]?(3[0-1]|0[1-9]|[1-9]|[1-2][0-9])$)")},
```

Comment:
IDK if this is intentional, but this regex:
- matches strings with different delimeters like `2024/12-01`
- does not match USA date format, that is, mm-dd-yyyy
check it yourself: https://regex101.com/r/MY1jCD/1

We can use `(\d{n})` to match a n digits and with `([-./])` we can capture a delimeter in a group and use it with `\group_number`.

I suggest using `^(?:(?!0000)(\d{4})([-.\/])(0[1-9]|1[0-2])\2(0[1-9]|[1-2][0-9]|3[0-1])|(0[1-9]|1[0-2])([-.\/])(0[1-9]|[1-2][0-9]|3[0-1])\6(?!0000)(\d{4}))$`
https://regex101.com/r/sOhFqg/1

Reply:
Our date type is based on [`boost::gregorian::date`](https://www.boost.org/doc/libs/1_86_0/doc/html/date_time/gregorian.html). So USA date format is not supported.
Also, the only proper way to check whether a string is a correct date is trying to create `boost::gregorian::date` object (which is done in `kDelimitedDateCheck` and `kUndelimitedDateCheck`). The regex is needed to quickly eliminate strings that are not dates and to pass anything that looks like `boost::gregorian::date`.

Should we use `^(\d{4})([-.\/]?)(1[0-2]|0[1-9]|[1-9])\2(3[0-1]|0[1-9]|[1-9]|[1-2][0-9])$`?
https://regex101.com/r/rF63Ft/1
The only difference from the original regex is that it checks for the delimiters.

Reply:
Understood. Yes, we should, thanks